### PR TITLE
Switch underlying package for GoogleSheets to googlesheets4 from googlesheets

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

Switch underlying package for GoogleSheets to googlesheets4 from googlesheets

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [x] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/google_sheets.R [hunk 1]
```diff
@@ -19,38 +22,55 @@ uploadGoogleSheet <- function(filepath, title, overwrite = FALSE){
...
+  googledrive::drive_set_token(token)
+  # To improve peformance, only get id, name and canEdit for each spreadsheet.
+  googledrive::drive_find(type = "spreadsheet", pageSize=1000, fields="files/id, files/name, files/capabilities/canEdit, nextPageToken")
 }
 
 #' API to get a list of available google sheets
 #' @export
 getGoogleSheetWorkSheetList <- function(tokenFileId="", title){
-  if(!requireNamespace("googlesheets")){stop("package googlesheets must be installed.")}
+  if(!requireNamespace("googlesheets4")){stop("package googlesheets must be installed.")}
```
Comment (hideaki) [inline R/google_sheets.R]:
```suggestion
  if(!requireNamespace("googlesheets4")){stop("package googlesheets4 must be installed.")}
```

File: R/google_sheets.R [hunk 2]
```diff
@@ -18,39 +21,66 @@ uploadGoogleSheet <- function(filepath, title, overwrite = FALSE){
...
 #' @param firstRowAsHeader - argument to control if you want to treat first row as header
 #' @param commentChar - treat the character as comment.
+#' @param tokenFileId - No longer used. It was kept for backward compatibility for the old Desktop Versions that don't handle OAuth token in server side.
 #' @param guessDataType - flag to tell if you want googlesheets::gs_read to guess column data type
+#' @param tzone - timezone
+#' @param id - ID off the sheet
 #' @export
-getGoogleSheet <- function(title, sheetName, skipNRows = 0, treatTheseAsNA, firstRowAsHeader = TRUE, commentChar, tokenFileId=NULL, guessDataType=TRUE, ...){
-  if(!requireNamespace("googlesheets")){stop("package googlesheets must be installed.")}
+getGoogleSheet <- function(title, sheetName, skipNRows = 0, treatTheseAsNA = NULL, firstRowAsHeader = TRUE, commentChar, tokenFileId=NULL, guessDataType=TRUE, tzone=NULL, id=NULL, ...){
```
Comment (hideaki) [inline R/google_sheets.R]:
```suggestion
getGoogleSheet <- function(title, sheetName, skipNRows = 0, treatTheseAsNA = NULL, firstRowAsHeader = TRUE, commentChar, tokenFileId = NULL, guessDataType = TRUE, tzone = NULL, id = NULL, ...){
```

File: R/google_sheets.R [hunk 3]
```diff
@@ -18,39 +21,66 @@ uploadGoogleSheet <- function(filepath, title, overwrite = FALSE){
...
+  if(!requireNamespace("googlesheets4")){stop("package googlesheets4 must be installed.")}
+  if(!requireNamespace("googledrive")){stop("package googledrive must be installed.")}
   if(!requireNamespace("stringr")){stop("package stringr must be installed.")}
 
   token <- getGoogleTokenForSheet(tokenFileId)
-  googlesheets::gs_auth(token)
-  gsheet <- googlesheets::gs_title(title)
+  googlesheets4::sheets_set_token(token)
+  googledrive::drive_set_token(token)
+  if(!is.null(id)) {
```
Comment (hideaki) [inline R/google_sheets.R]: Please add a comment on why we do this switch.

File: R/google_sheets.R [hunk 4]
```diff
@@ -18,39 +21,66 @@ uploadGoogleSheet <- function(filepath, title, overwrite = FALSE){
...
+  } else {
+    gsheet <- googledrive::drive_get(title)
+  }
   col_types <- NULL
   if(!guessDataType) {
     # if guessDataType is FALSE, use character as the default column data type.
     col_types <- c(.default="c")
   }
-  df <- gsheet %>% googlesheets::gs_read(ws = sheetName, skip = skipNRows, na = treatTheseAsNA, col_names = firstRowAsHeader, comment = commentChar, col_types = col_types)
+  if(!is.null(treatTheseAsNA)) {
```
Comment (hideaki) [inline R/google_sheets.R]: Add a comment on why this is necessary.

File: R/google_sheets.R [hunk 5]
```diff
@@ -18,39 +21,66 @@ uploadGoogleSheet <- function(filepath, title, overwrite = FALSE){
...
+  if(!requireNamespace("googlesheets4")){stop("package googlesheets4 must be installed.")}
+  if(!requireNamespace("googledrive")){stop("package googledrive must be installed.")}
+
   token = getGoogleTokenForSheet(tokenFileId)
-  googlesheets::gs_auth(token)
-  sheet <- googlesheets::gs_title(title)
-  googlesheets::gs_ws_ls(sheet)
+  googlesheets4::sheets_set_token(token)
+  googledrive::drive_set_token(token)
+  if(id != "") {
```
Comment (hideaki) [inline R/google_sheets.R]: Add comment on the background of this.

File: R/google_sheets.R (hunk 3 above)
Comment (hidekoji) [reply R/google_sheets.R]: Added comment.

File: R/google_sheets.R (hunk 4 above)
Comment (hidekoji) [reply R/google_sheets.R]: Added comment.

File: R/google_sheets.R (hunk 5 above)
Comment (hidekoji) [reply R/google_sheets.R]: Added comment.

**Pull request merged**